vcd/resource_vcd_ip_space_ip_allocation.go

```diff
@@ -0,0 +1,304 @@
+			Description: d.Get("description").(string),
+			Value:       ipSpaceAllocation.IpSpaceIpAllocation.Value,
+		}
+
+		_, err = ipSpaceAllocation.Update(ipAllocationUpdate)
+		if err != nil {
+			return diag.Errorf("error updating IP Space IP Allocation after creation: %s", err)
+		}
+	}
+
+	return resourceVcdIpAllocationRead(ctx, d, meta)
+}
+
+func resourceVcdIpAllocationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("[TRACE] IP Space IP Allocation update initiated")
+
+	vcdClient := meta.(*VCDClient)
+
+	orgId := d.Get("org_id").(string)
+	ipSpaceId := d.Get("ip_space_id").(string)
+	org, err := vcdClient.GetOrgById(orgId)
+	if err != nil {
+		return diag.Errorf("error getting Org by ID: %s", err)
+	}
+
+	ipAllocation, err := org.GetIpSpaceAllocationById(ipSpaceId, d.Id())
+	if err != nil {
+		return diag.Errorf("error retrieving IP Allocation: %s", err)
+	}
+
+	if d.HasChange("usage_state") || d.HasChange("description") {
+		ipAllocation.IpSpaceIpAllocation.UsageState = d.Get("usage_state").(string)
+		ipAllocation.IpSpaceIpAllocation.Description = d.Get("description").(string)
+		_, err = ipAllocation.Update(ipAllocation.IpSpaceIpAllocation)
+		if err != nil {
+			return diag.Errorf("error updating IP Space IP Allocation: %s", err)
+		}
+	}
+
+	return resourceVcdIpAllocationRead(ctx, d, meta)
+}
+
+func resourceVcdIpAllocationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf("[TRACE] IP Space IP Allocation read initiated")
+
+	vcdClient := meta.(*VCDClient)
+	orgId := d.Get("org_id").(string)
+	ipSpaceId := d.Get("ip_space_id").(string)
+
+	org, err := vcdClient.GetOrgById(orgId)
+	if err != nil {
+		return diag.Errorf("error getting Org by id: %s", err)
+	}
+
+	ipAllocation, err := org.GetIpSpaceAllocationById(ipSpaceId, d.Id())
+	if err != nil {
+		return diag.Errorf("error getting IP Space IP Allocation: %s", err)
+	}
+
+	dSet(d, "description", ipAllocation.IpSpaceIpAllocation.Description)
+	dSet(d, "type", ipAllocation.IpSpaceIpAllocation.Type)
+	if ipAllocation.IpSpaceIpAllocation.OrgRef != nil {
+		dSet(d, "org_id", ipAllocation.IpSpaceIpAllocation.OrgRef.ID)
+	}
+	if ipAllocation.IpSpaceIpAllocation.UsedByRef != nil {
+		dSet(d, "used_by_id", ipAllocation.IpSpaceIpAllocation.UsedByRef.ID)
+	}
+	dSet(d, "allocation_date", ipAllocation.IpSpaceIpAllocation.AllocationDate)
+	dSet(d, "usage_state", ipAllocation.IpSpaceIpAllocation.UsageState)
+	dSet(d, "ip_address", ipAllocation.IpSpaceIpAllocation.Value)
+
+	// When IP Prefix is allocated, the returned value is in CIDR format (e.g. 192.168.1.0/24), and
+	// although it can be split using Terraform native functions, we're adding a convenience layer for
+	// users by splitting this address into IP and prefix length
+	if ipAllocation.IpSpaceIpAllocation.Type == "IP_PREFIX" {
+		splitCidr := strings.Split(ipAllocation.IpSpaceIpAllocation.Value, "/")
+		if len(splitCidr) == 2 {
```

Comment:
Added.
 We will be able to check what exactly was returned in the error code, but logging that additionally now (I don't want to "fail" just because of that as one could still use the `ip_address` field which is a value directly coming from the API. At least someone doesn't have to wait for bugfix if this happens.